src/pages/HomePage.js

```diff
@@ -1,15 +1,35 @@
-import React, { useContext } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 
-import useListToken from '../useListToken';
+import { getCurrentToken } from '../useListToken';
 import { ListContext } from '../listContext';
 import List from './List';
 import Welcome from '../components/Welcome';
+import Loading from '../components/Loading';
 
 const HomePage = () => {
-  const { token } = useListToken();
-  const { validToken } = useContext(ListContext);
+  const { initializeList, shoppingList } = useContext(ListContext);
+  const [loading, setLoading] = useState(true);
 
-  return <>{validToken(token) ? <List /> : <Welcome />}</>;
+  const [existingList, setExistingList] = useState(
+    initializeList(getCurrentToken()),
+  );
+  const confirmListStatus = React.useCallback(() => {
+    setExistingList(shoppingList.length > 0);
+    setLoading(false);
+  }, [shoppingList.length]);
+
+  useEffect(() => {
+    confirmListStatus();
+    console.log('token from HomePage useEffect', getCurrentToken());
+    console.log('existingList value from HomePage useEffect', existingList);
+  }, [confirmListStatus, existingList]);
+
+  return (
+    <>
+      {loading && <Loading />}
+      {existingList ? <List /> : !loading && !existingList && <Welcome />}
```

Comment:
@stevelikesmusic How do I get the list to show without the Welcome component flashing in?

Reply:
I think I get what you're trying to do. You're on the right track @prophen! Wish I had a short concise answer for this one :joy:

There's a long-term fix and a short-term bandaid. The bandaid is the easiest one. That would be to check if we have a token or not with `useListToken`. If we have a valid token, then we technically have a list—though it may be empty. This will be synchronous and we won't need to reach for `useEffect` to set loading state. It would also fix a bug with the current approach—if I have a token but no items, we think I need to create a new list. 

The long-term fix would be to either:
1. Use `listContext` for all things list related. That would mean getting rid of `FirebaseCollection` usage because `listContext` would be managing everything. That would also mean us maintaining loading and error states ourselves. 
1. Only use `listContext` for mutating the list. It wouldn't keep the list in memory. Instead, we'd use `FirebaseCollection` when we want to read data and pass in the list to `listContext` for dupe checking and such. 

We're in a weird state now with two lists getting passed around and inconsistent loading state. Both approaches have their tradeoffs. 

At the least, I might remove the loading state in `HomePage` and just always render `List` since it shows loading state already for loading the shopping list. Then we can redirect to `Welcome` if we don't have a token—which could be its own route.